app/src/app/admin/abstractEntitiesController.ts

```diff
@@ -86,6 +95,20 @@ namespace app.admin {
             });
         }
 
+        /**
+         * Returns the user to the previous state.
+         */
+        public cancel():void {
+
+            if(this.previousState.abstract) {
```

Comment:
Is there a better way to check if the last state is invalid? E.g. The user clicked on a direct link for an article.